apps/explorer/src/ui/InternalLink.tsx

```diff
@@ -1,21 +1,27 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { formatAddress } from '../utils/stringUtils';
+import { formatAddress, truncate } from '../utils/stringUtils';
 
 import { Link, type LinkProps } from '~/ui/Link';
 
 interface BaseInternalLinkProps extends LinkProps {
     noTruncate?: boolean;
 }
 
+const TRUNCATE_LENGTH = 10;
+
 function createInternalLink<T extends string>(base: string, propName: T) {
     return ({
         [propName]: id,
         noTruncate,
         ...props
     }: BaseInternalLinkProps & Record<T, string>) => {
-        const truncatedAddress = noTruncate ? id : formatAddress(id);
+        const truncatedAddress = noTruncate
+            ? id
+            : base === 'transaction'
```

Comment:
Instead of using the base, can you have this function take an optional arg like `formatter` that you pass when creating the internal link for transactions